src/widgets/reflow.rs

```diff
@@ -34,11 +34,15 @@ where
     /// The given, unprocessed lines
     input_lines: O,
     max_line_width: u16,
-    wrapped_lines: Option<IntoIter<Vec<StyledGrapheme<'a>>>>,
     current_alignment: Alignment,
     current_line: Vec<StyledGrapheme<'a>>,
     /// Removes the leading whitespace from lines
     trim: bool,
+
+    wrapped_lines: VecDeque<Vec<StyledGrapheme<'a>>>,
+    pending_word: Vec<StyledGrapheme<'a>>,
+    pending_whitespace: VecDeque<StyledGrapheme<'a>>,
+    pending_line_pool: Vec<Vec<StyledGrapheme<'a>>>,
```

Comment:
Would be helpful to document how each of these is used / and how they fit into the algorithm. Reflow is mostly magic (it has traditionally had some high complexity, which is perhaps lower now than it has been in the past). But I don't pretend to (or even try to) understand it in its current state.

Reply:
`wrapped_lines` usage hasn't changed (the type just went from an iterator to a buffer) and I added a comment to the others indicating they're just caches and don't actually hold any data. I'm mainly pulling vecs that were defined inside `process_input` to the struct so their allocations are reused.

Reply:
Yeah, my problem is every time I jump into this code it's something that I have to relearn from scratch - off the top of my head I couldn't describe the algorithm in detail except it statefully iterates over the lines and puts them into lines (which is sort of definitional rather than algorithmic so 🤷).

Reply:
Lol, fair enough.